### DISCUSSION TITLE
Page numbers in export to PDF

### DISCUSSION THREAD

**Discussion by JMalton** [jgraph/drawio#4710] [Q&A]
I use a background sheet with my drawings, which prints and fills a title block, global data, sheet frame, revision info, date, page number, etc on each sheet.  That sheet is then locked as _Background_ (which doesn't seem to make much difference, as it gets exported anyway), and assigned as the background sheet to every other sheet.

On the background sheet, there are two fields (in my title block section), with _Placeholders_ enabled, and containing the text

- **%date{yyyy-mm-dd}%**
- **%pagenumber%** of **%pagecount%**

The **%date%** works fine.

The **%pagecount%** is wrong, of course, because the locked _Background_ page is unfortunately included in the page count

The **%pagenumber%** works fine in the app, but it _DOESN'T WORK ON PDF EXPORT_.  Rather, on export, each sheet's title block gets the page number of the sheet that was currently in focus when the _Export_ function was invoked.

Has anyone found a workaround for this?

**Comment by euan-cowie** [comment]
Have this same issue

**Comment by WindowPlaceBrain** [comment]
We also got that problem, would be interested in a solution to pagecount and a fix for the pagenumber.

**Comment by core-engineering** [comment]
Hi,

I got exactly the same behavior for %pagenumber%. Workaround for me : printing in pdf !

**Comment by maximeLeurent** [comment]
Unfortunately for me, it doesn’t work: I get the same issue where the page count equals the active page at the moment of printing.
Moreover, it has lost all the links between pages, something I was able to keep with the normal export functionality.

I found another workaround:
Export to HTML → open the HTML file → print to PDF.
This way, I kept both the links and the page numbers.
I just had to add small items in the top-left and bottom areas of my background page; otherwise, the size of each page was recalculated based on the content.

**Comment by alderg** [comment]
Links and pagecount/-numbers for me. Tested in v29.2.9 with Chrome 143.0.7499.170 on macOS 26.1. Please include browser, operating system, versions and exact reproduction steps.

BTW: You can now use %pagecount-1% etc, see https://www.drawio.com/doc/faq/predefined-placeholders

**Comment by DazB** [comment]
I also have this issue on the draw.io standalone app (v29.6.6).
 
I can see the correct %pagenumber% placeholder when I have the page open in draw.io, but when I Export as PDF, all I see is the same page number (1) on every page. 

If I Print as PDF, it shows correctly on PDF. However it is printed rotated, so I'd prefer if Export as PDF worked correctly.

**Comment by alderg** [comment]
This will be fixed in the next release (9887818b6).
